### PR TITLE
fix: adding backward compatibility notes to wire-reform rfc.

### PR DESCRIPTION
**Pull request by jodarove**

### CODE REVIEW

File: text/0000-wire-reform.md [hunk 1]
```diff
@@ -100,6 +100,9 @@ There exist a few restrictions and ambiguities with the IDL for the config objec
 * there will be no identity for inline JSON objects when assigned to a property in the config object, e.g.: `@wire(foo, { x: { y: 1 } })` where the value of `x` will be computed every time, instead of cached per instance or per class as today.
 * there will be identity preserved when assigning a reference values in the config object, e.g.: `@wire(foo, { x: someValue })` where the value of `x` will be a reference to `someValue` during the class declaration.
 * every time that `adapter.update()` is invoked, a new config object will be provided as a first argument, no identity is preserved in this case.
+* `adapter.update()` will be invoked initially regardless of the value of the config. e.g.: `@wire(foo, { x: $foo })` and `this.foo` resolves to `undefined`.
```
Comment (caridy) [inline text/0000-wire-reform.md:103]:
```suggestion
* `adapter.update()` will be invoked initially regardless of the value of the config. e.g.: `@wire(foo, { x: $foo })` will invoke the update even if `this.foo` resolves to `undefined`.
```

File: text/0000-wire-reform.md [hunk 2]
```diff
@@ -100,6 +100,9 @@ There exist a few restrictions and ambiguities with the IDL for the config objec
...
+* on any change to the values used to generate a wire configuration, `adapter.update()` will be called with the new config object, even if results in the same config values.
+* since now the underlying reactive tracking for wire configuration is using the same reactivity system as LWC, the wire protocol can’t track changes on a wire configuration that depends on class expandos.
```
Comment (caridy) [inline text/0000-wire-reform.md:105]:
```suggestion
* the reactive tracking for wire configuration can’t track changes on a wire configuration that depends on component's instance "expandos", it only reacts to changes on class' declared fields.
```

File: text/0000-wire-reform.md [hunk 3]
```diff
@@ -100,6 +100,9 @@ There exist a few restrictions and ambiguities with the IDL for the config objec
 * there will be no identity for inline JSON objects when assigned to a property in the config object, e.g.: `@wire(foo, { x: { y: 1 } })` where the value of `x` will be computed every time, instead of cached per instance or per class as today.
 * there will be identity preserved when assigning a reference values in the config object, e.g.: `@wire(foo, { x: someValue })` where the value of `x` will be a reference to `someValue` during the class declaration.
 * every time that `adapter.update()` is invoked, a new config object will be provided as a first argument, no identity is preserved in this case.
+* `adapter.update()` will be invoked initially regardless of the value of the config. e.g.: `@wire(foo, { x: $foo })` and `this.foo` resolves to `undefined`.
+* on any change to the values used to generate a wire configuration, `adapter.update()` will be called with the new config object, even if results in the same config values.
```
Comment (caridy) [inline text/0000-wire-reform.md:104]:
```suggestion
* `adapter.update()` will be called when the reactive mechanism detects a mutation on a value that was used to compute the config for a wire declaration, even if result of such computation produces the same config that was produced before. It is a responsibility of the adapter to dedupe such calls.
```

**Comment by jodarove** [comment]
great suggestions, committed them.

**Pull request merged**